File: ios/CommonParams.swift
import ExpoModulesCore

struct CommonParams: Record {
    @Field
    var chainId: Int?
    
    @Field
    var rpcUrl: String?
    
    @Field
    var bundlerUrl: String?
    
    @Field
    var config: [String: String]?
    
    @Field
    var address: String?
    
    @Field
    var paymasterUrl: String?
    
    @Field
    var signer: [String: Any]?

}

extension CommonParams {
    func verify() throws {
        if rpcUrl == nil {
            throw Rtn4337Exception("rpcUrl is required")
        }
        if bundlerUrl == nil {
            throw Rtn4337Exception("bundlerUrl is required")
        }
        if config == nil {
            throw Rtn4337Exception("config is required")
        }
    }
}

File: android/src/main/java/io/cometh/rtn4337/types/CommonParams.kt
package io.cometh.rtn4337.types

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class CommonParams : Record {
    @Field
    val chainId: Int? = null

    @Field
    val rpcUrl: String? = null

    @Field
    val bundlerUrl: String? = null

    @Field
    val config: Map<String, String>? = null

    @Field
    val address: String? = null

    @Field
    val paymasterUrl: String? = null

    @Field
    val signer: Map<String, String>? = null

}

fun CommonParams.verifyMandatoryUrls() {
    requireNotNull(rpcUrl) { "rpcUrl is required" }
    requireNotNull(bundlerUrl) { "bundlerUrl is required" }
    requireNotNull(config) { "config is required" }
}

